File: 2025-09-16_find_the_highest_altitude.java
// Problem: Find The Highest Altitude
// Link: https://leetcode.com/problems/find-the-highest-altitude/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
    public int largestAltitude(int[] gain) {
        int altitude=0;
        int maxAltitude=0;

        for(int g:gain){
            altitude+=g;
            maxAltitude=Math.max(maxAltitude,altitude);
        }
        return maxAltitude;
    }
}

File: 2025-09-16_find_the_highest_altitude.py
// Problem: Find the highest Altitude
// Link: https://leetcode.com/problems/find-the-highest-altitude/description/?envType=study-plan-v2&envId=leetcode-75

class Solution(object):
    def largestAltitude(self, gain):
        """
        :type gain: List[int]
        :rtype: int
        """
        altitude=0
        maxAltitude=0

        for g in gain:
            altitude+=g
            maxAltitude=max(maxAltitude,altitude)
        
        return maxAltitude